docs/source/format/how_manager.rst

```diff
@@ -0,0 +1,204 @@
+================================================
+
+.. note:: This document focuses on drivers/applications using the C API
+          definitions in adbc.h.  That means C/C++/Python/Ruby, and possibly
+          C#/Go.
+
+When an application calls a function like
+:cpp:func:`AdbcStatementExecuteQuery`, how does it "know" what function in
+which driver to actually call?
+
+This can happen in a few ways.  In the simplest case, the application links to
+a single driver, and directly calls ADBC functions explicitly defined by the
+driver:
+
+.. figure:: DriverDirectLink.mmd.svg
+
+   In the simplest case, an application directly links to the driver and calls
+   ADBC functions.
+
+This doesn't work with multiple drivers, or applications that don't/can't link
+directly to drivers (think dynamic loading, perhaps in a language like
+Python).  For this case, ADBC provides a table of function pointers
+(:cpp:struct:`AdbcDriver`), and a way to request this table from a driver.
+Then, the application proceeds in two steps.  First, it dynamically loads a
+driver and calls an entrypoint function to get the function table:
+
+.. figure:: DriverTableLoad.mmd.svg
+
+   Now, the application asks the driver for a table of functions to call.
+
+Then, the application uses the driver by calling the functions in the table:
+
+.. figure:: DriverTableUse.mmd.svg
+
+   The application uses the table to call driver functions.  This approach
+   scales to multiple drivers.
+
+Dealing with the table, however, is messy.  So the overall recommended
+approach is to use the ADBC driver manager.  This is a library that pretends
+to be a single driver that can be linked to and used "like normal".
+Internally, it loads the table of function pointers and tracks which
+database/connection/statement objects need which "actual" driver, making it
+easy to dynamically load drivers at runtime and use multiple drivers from the
+same application:
+
+.. figure:: DriverManagerUse.mmd.svg
+
+   The application uses driver manager to "feel like" it's just using a single
+   driver.  The driver manager handles the details behind the scenes.
+
+In More Detail
+==============
+
+The adbc.h header ties everything together.  It is the abstract API
+definition, akin to interface/trait/protocol definitions in other languages.
+C being C, however, all it consists of is a bunch of function prototypes and
+struct definitions without any implementation.
+
+A driver, at its core, is just a library that implements those function
+prototypes in adbc.h.  Those functions may be implemented in C, or they can be
+implemented in a different language and exported through language-specific FFI
+mechanisms.  For example, the Go and C# implementations of ADBC can both
+export drivers to consumers who expect the C API definitions.  As long as the
+definitions in adbc.h are implemented somehow, then the application is
+generally none the wiser when it comes to what's actually underneath.
```

Comment:
Should we provide a description of a minimal driver implementation? Or the minimum amount necessary to implement?

Reply:
I don't think it's relevant here.

Reply:
Elsewhere in the document?